File: modules/ocr/ios/ScanViewModule.swift
import ExpoModulesCore

// Registers the native real-time scanner view. The view owns an AVCaptureSession
// that feeds QR (metadata output) and OCR (video frames → Vision) at the same
// time; both surface to JS as events, where scanInput.ts arbitrates a lock.
public class ScanViewModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ScanView")

    View(ScanCameraView.self) {
      Events("onBarcode", "onText")

      // Whether the capture session runs at all (tie to screen focus). Stopping
      // it releases the camera when the tab is backgrounded.
      Prop("active") { (view: ScanCameraView, active: Bool) in
        view.setActive(active)
      }

      // Whether detections are emitted. Preview keeps running while false, so a
      // locked result can freeze scanning without tearing down the camera.
      Prop("scanning") { (view: ScanCameraView, scanning: Bool) in
        view.scanning = scanning
      }
    }
  }
}

File: modules/ocr/android/src/main/java/expo/modules/ocr/ScanViewModule.kt
package expo.modules.ocr

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

// Android counterpart of the iOS ScanViewModule (modules/ocr/ios). Registers the
// native "ScanView" — a CameraX preview that runs ML Kit OCR (Latin) + QR on the
// live frames and emits onText/onBarcode, matching the iOS view's contract so
// src/scanInput.ts + Scan.tsx arbitrate identically across platforms.
class ScanViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ScanView")

    View(ScanCameraView::class) {
      Events("onBarcode", "onText")

      // Run the capture session (tie to screen focus). False releases the camera.
      Prop("active") { view: ScanCameraView, active: Boolean ->
        view.setActive(active)
      }

      // Emit detections. Preview keeps running while false, so a locked result
      // can freeze scanning without tearing down the camera.
      Prop("scanning") { view: ScanCameraView, scanning: Boolean ->
        view.scanning = scanning
      }
    }
  }
}
